core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryServerLiveAttributesProvider.java

```diff
@@ -0,0 +1,87 @@
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.telemetry;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.sonarsource.sonarlint.core.repository.config.ConfigurationRepository;
+import org.sonarsource.sonarlint.core.repository.connection.ConnectionConfigurationRepository;
+import org.sonarsource.sonarlint.core.repository.connection.SonarCloudConnectionConfiguration;
+import org.sonarsource.sonarlint.core.repository.rules.RulesRepository;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.telemetry.TelemetryServerLiveAttributesDto;
+import org.sonarsource.sonarlint.core.rules.RulesService;
+
+@Named
+@Singleton
+public class TelemetryServerLiveAttributesProvider {
+
+  private final ConfigurationRepository configurationRepository;
+  private final ConnectionConfigurationRepository connectionConfigurationRepository;
+  private final RulesService rulesService;
+  private final RulesRepository rulesRepository;
+
+  public TelemetryServerLiveAttributesProvider(ConfigurationRepository configurationRepository,
+    ConnectionConfigurationRepository connectionConfigurationRepository,
+    RulesService rulesService, RulesRepository rulesRepository) {
+    this.configurationRepository = configurationRepository;
+    this.connectionConfigurationRepository = connectionConfigurationRepository;
+    this.rulesService = rulesService;
+    this.rulesRepository = rulesRepository;
+  }
+
+  public TelemetryServerLiveAttributesDto getTelemetryServerLiveAttributes() {
+    var allBindings = configurationRepository.getConfigScopeIds().stream()
+      .map(configurationRepository::getBindingConfiguration)
+      .filter(bindingConfiguration -> bindingConfiguration != null && bindingConfiguration.getConnectionId() != null)
+      .collect(Collectors.toList());
+
+    var usesConnectedMode = !allBindings.isEmpty();
+
+    var usesSonarCloud = allBindings.stream()
+      .anyMatch(binding -> connectionConfigurationRepository.getConnectionById(binding.getConnectionId()) instanceof SonarCloudConnectionConfiguration);
```

Comment:
I don't like this "instanceOf". We could maybe add a `isSonarCloud` attribute on the `BoundScope`? I remember it would have been useful for another use case.